napari/settings/_base.py

```diff
@@ -366,6 +367,29 @@ def config_file_settings_source(
     sources = list(getattr(settings.__config__, 'sources', []))
     if config_path:
         sources.append(config_path)
+        # check for previous version directory, but only if after 0.4.17
+        path_version = version.parse(str(Path(config_path).parts[-2]))
+        if isinstance(
+            path_version, version.Version
+        ) and path_version > version.parse('0.4.17'):
+            *v, rev = str(Path(config_path).parts[-2]).split('.')
+            prev_v = ".".join(v) + '.' + str(int(rev) - 1)
```

Comment:
this will not work with `0.5.0`

Reply:
Good catch. 
I wonder if it should work? like is 0.5.0 going to be so major that you don't want to import previous settings? (I think it won't error as written, it will just return not found warning further down.)

Reply:
> I wonder if it should work?

It should work and napari should develop mechanism of migration settings from previous version.

Reply:
Hmm, I can't think of a way to a priori know what the last <rev> of a previous <version> was. Like for 0.5.0 the previous might be 0.4.17—unlikely, but possible—0.4.18—more likely—or 0.4.19. But for 0.6.0? 🤷‍♂️  
I guess a different approach is to actually list the contents of the napari appdir and look for directories that are `instance version.Version` and go off that?

Reply:
> I guess a different approach is to actually list the contents of the napari appdir and look for directories that are `instance version.Version` and go off that?

yes

Reply:
Ok, i got that implemented:
Look for dirs in `napari` dir, check if they are versions, then use the most recent to load settings.
If there are no dirs that are versions then just use `napari` dir.